ui/components/app/signature-request/signature-request.component.js

```diff
@@ -62,18 +64,23 @@ export default class SignatureRequest extends PureComponent {
   render() {
     const {
       fromAccount,
-      txData: {
-        msgParams: { data, origin, version },
-        type,
-      },
+      txData,
       cancel,
       sign,
       isLedgerWallet,
       hardwareWalletRequiresConnection,
+      subjectMetadata,
     } = this.props;
+    const {
+      msgParams: { data, origin, version },
+      type,
+    } = txData;
     const { address: fromAddress } = fromAccount;
-    const { message, domain = {}, primaryType, types } = JSON.parse(data);
```

Comment:
Huh. I had not realized that the domain information isn't present anywhere :thinking: It is effectively a part of the message payload, so I would expect it to be rendered there. But it is not.

Maybe we should pass `domain` along to the `Message` component to be displayed alongside `data`.